### ISSUE TITLE
react-server-cli wishlist

### ISSUE THREAD

**Issue by doug-wade** [redfin/react-server#421]
Pulled from the react-server-cli README

Here are a few of the things on the unordered wishlist to add to `react-server-cli`:

* Default to https, with http as a special case. If no cert & key specified, make a self-signed cert/key combo.
* Ability to output a completely static site, if the user wants to forgo server-side rendering.
* Ability to forgo server rendering in development mode.
* A programmatic API.
* Automatic compilation of SASS.
* Ability to opt out of Babel compilation.
* Inclusion of CSS Modules
* Help with proxying API endpoints.

**Comment by axelson** [comment]
Support for SASS was added in #543 

I'm really looking forward to CSS Modules

**Comment by patrick-mcdougle** [comment]
hot reloading in dev would be really swell.

**Comment by breeswish** [comment]
Are there any plans to make SASS optional? There could be a lot of things to compile besides SASS, e.g. Stylus, TypeScript, etc. It may be better to implement a plug-in system to include and install SASS as needed.

**Comment by doug-wade** [comment]
@breeswish you can already add [custom webpack loaders](https://react-server.io/docs/guides/react-server-cli)

**Comment by breeswish** [comment]
Yeah I know. I'm suggesting a plugin system so that undesired components won't be included.
